services-api/src/main/java/io/scalecube/services/Reflect.java

```diff
@@ -324,4 +349,9 @@ public static void setField(Field field, Object object, Object value)
   public static boolean isService(Class<?> type) {
     return type.isAnnotationPresent(Service.class);
   }
+
+  public static boolean isAuth(Method method) {
+    return method.isAnnotationPresent(Auth.class)
+        || method.getDeclaringClass().isAnnotationPresent(Auth.class);
+  }
```

Comment:
`@Auth` on class or method turns a method (or all methods if on class) to _auth_ one. Ok.
But can a method (yes, one method)  be _auth_ one without defining `@Auth` annotation on the method? For example client didn't put auth annotation but put a `@Principal` on a parameter; what to do in this case? Formally he didn't put auth annotation so method is not _auth_ one, but he has a principal, so he wants to be auth.

Reply:
One can put `@Auth` on method just to allow this method to be verified with the `Authenticator` and no further action (e.g. principal verification / access control) is needed.
If someone puts `@Principal` on a parameter without `@Auth` (neither method nor class) **I think** it should imply `@Auth` on that method.

Reply:
Added additional validation https://github.com/scalecube/scalecube-services/pull/624/commits/b8b37531fd0371cd1708f10b6b6687c6f1f18b0a

Reply:
@aharonha 
What you say makes sense. But let's not do it from start, unless we face this problem down the road: when somebody start to use this and start to complain on us   _hey why putting `@Principal` is not enough?_